File: src/gamesbyexample/shiningcarpet.py
"""Shining Carpet, by Al Sweigart al@inventwithpython.com
Displays a tessellation of the carpet pattern from The Shining.
This and other games are available at https://nostarch.com/XX
Tags: tiny, beginner, artistic"""
__version__ = 0
# Set up the constants:
X_REPEAT = 6  # How many times to tessellate horizontally.
Y_REPEAT = 4  # How many times to tessellate vertically.

for i in range(Y_REPEAT):
    print(r'_ \ \ \_/ __' * X_REPEAT)
    print(r' \ \ \___/ _' * X_REPEAT)
    print(r'\ \ \_____/ ' * X_REPEAT)
    print(r'/ / / ___ \_' * X_REPEAT)
    print(r'_/ / / _ \__' * X_REPEAT)
    print(r'__/ / / \___' * X_REPEAT)

File: javascript_games/shiningcarpet.js
/*
Shining Carpet, by Al Sweigart al@inventwithpython.com
Displays a tessellation of the carpet pattern from The Shining.
*/

'use strict';

// Set up the constants:
const X_REPEAT = 6;  // How many times to tessellate horizontally.
const Y_REPEAT = 4;  // How many times to tessellate vertically.

for (let y = 0; y < Y_REPEAT; y++) {
    console.log('_ \\ \\ \\_/ __'.repeat(X_REPEAT));
    console.log(' \\ \\ \\___/ _'.repeat(X_REPEAT));
    console.log('\\ \\ \\_____/ '.repeat(X_REPEAT));
    console.log('/ / / ___ \\_'.repeat(X_REPEAT));
    console.log('_/ / / _ \\__'.repeat(X_REPEAT));
    console.log('__/ / / \\___'.repeat(X_REPEAT));
}
